docs/Dev_eXo_Addons.rst

```diff
@@ -1139,11 +1150,188 @@ following:
+
+.. note:: The term ``point`` here is abstract and can represent anything meanfingful for the rewarding program.
+          Giving existing examples :
+            - In gamification, point is a gamification point
+            - In Kudos, point is a kudos
+          You can, for example, add a custom reward program ``training`` where ``point`` is the average score obtained in exams.
+
+When the Rewards package administrator will pay rewards to users for a period of time (each month for example),
+the rewards plugins will be used to retrieve earned points for each user having a wallet.
+The earned points will be converted into an amount of tokens to send to each user.
+
+For example, the `Kudos reward plugin` is used to return the number of Kudos received by a user during a selected period of time.
+Administrators can configure the token value for each earned Kudos (or a total budget), so the rewarding engine can compute individual amounts of tokens for each user.
+
+Developing a custom Reward plugin, requires:
+
+* to write a java Class that extends `RewardPlugin <https://github.com/exoplatform/wallet/blob/develop/wallet-api/src/main/java/org/exoplatform/addon/wallet/reward/api/RewardPlugin.java>`__
+
+   .. code:: java
+   
+      public class CustomRewardPlugin extends RewardPlugin {
+        /**
+         * Retrieves earned points for identities in a selected period of time
+         * 
+         * @param identityIds identity ids of wallets to consider in computation
+         * @param startDateInSeconds start timestamp in seconds of reward period
+         * @param endDateInSeconds end timestamp in seconds of reward period
+         * @return a {@link Map} of identity ID with the sum of tokens to send as
+         *         reward
+         */
+        public Map<Long, Double> getEarnedPoints(Set<Long> identityIds, long startDateInSeconds, long endDateInSeconds) {
+          // compute earned points per identityId. This example will return
+          the same number of points per user.
+          // Hint: The user social profile can be retrieved using
+          // org.exoplatform.social.core.manager.IdentityManager.getIdentity(String.valueOf(identityId), true)
+          double earnedPoints = 2d; // constant for all identities (dummy value)
+          return identityIds.stream().collect(Collectors.toMap(Function.identity(), id -> earnedPoints));
+        }
+      }
+
+* to declare the component plugin in a :ref:`Portal extension <PLFDevGuide.eXoAdd-ons.PortalExtension>` configuration file:
+
+   .. code:: xml
+
+      <external-component-plugins>
+        <target-component>org.exoplatform.addon.wallet.reward.service.RewardSettingsService</target-component>
+        <component-plugin>
+          <!-- Reward plugin name: must be unique  -->
+          <name>custom</name>
+          <set-method>registerPlugin</set-method>
+          <!-- FQN of plugin class  -->
+          <type>org.example.CustomRewardPlugin</type>
```

Comment:
You should declare the package in the defined CustomRewardPlugin for coherence

Reply:
done